### PR TITLE
Feature: Message

### PR DESCRIPTION
**Pull request by alanensastegui**
Added a recipient field to direct messages. Also updated MessageServlet and DataStore accordingly to handle the recipient field. Actual functionality of the direct messaging feature is unchanged.

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Message.java [hunk 1]
```diff
@@ -25,20 +25,26 @@
   private String user;
   private String text;
   private long timestamp;
+  private String recipient;
 
   /**
    * Constructs a new {@link Message} posted by {@code user} with {@code text} content. Generates a
```
Comment (Tetheta) [inline src/main/java/com/google/codeu/data/Message.java]: I know the docs you're following don't do this, but you should update the comments here to reflect the new variable you're adding.

File: src/main/java/com/google/codeu/data/Message.java [hunk 2]
```diff
@@ -25,20 +25,26 @@
...
    */
-  public Message(String user, String text) {
-    this(UUID.randomUUID(), user, text, System.currentTimeMillis());
+  public Message(String user, String text, String recipient) {
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient);
   }
 
-  public Message(UUID id, String user, String text, long timestamp) {
+  /**
+   * Constructs a new {@link Message} posted by {@code user}, of the given
```
Comment (Tetheta) [inline src/main/java/com/google/codeu/data/Message.java:44]: Where does recipient fit into this sentence/jsdoc? Same as above basically

**Review by Tetheta — CHANGES_REQUESTED** [review]
I believe the "summary jsdoc" basically just wants a bit of info about what the function does besides the return type. If it doesn't go away after a bit of tweaking don't worry about it.

Do fix the indentation and add to the comments a bit though. I know they don't have you doing it in the tutorial/docs but it's good practice to keep comments up to date (nothing is worse than a misleading comment in the future)

File: src/main/java/com/google/codeu/data/Message.java (hunk 1 above)
Comment (alanensastegui) [reply src/main/java/com/google/codeu/data/Message.java]: Updated the method comment to include recipient!

File: src/main/java/com/google/codeu/data/Message.java (hunk 2 above)
Comment (alanensastegui) [reply src/main/java/com/google/codeu/data/Message.java:44]: Also updated this method comment to include recipient!

**Comment by alanensastegui** [comment]
> I believe the "summary jsdoc" basically just wants a bit of info about what the function does besides the return type. If it doesn't go away after a bit of tweaking don't worry about it.
> 
> Do fix the indentation and add to the comments a bit though. I know they don't have you doing it in the tutorial/docs but it's good practice to keep comments up to date (nothing is worse than a misleading comment in the future)

**Review by Tetheta — APPROVED** [review]

**Comment by Tetheta** [comment]
@1alan98  Looks good, feel free to merge

**Pull request merged**